### PR TITLE
docs: add dedicate page for using secret store within Azure Functions

### LINKED ISSUE #301: Add extra link on secret store index page to 'How to use secret store in Azure Functions'
> We should have some simple documentation on how to use / integrate the ISecretProvider in an Azure Function project.

>I know that we consider the Arcus Templates as documentation / getting started as well, but I think it is not sufficient in this case.
(For instance, people that already have an existing Function who want to make use of the SecretProvider, or if you started a Function project but didn't use the template because you're not going to use a Http trigger (AFAIK, there only exists a template for functions which creates a Function that is triggered via HTTP).

We should add an extra link to the secret store index page for more visibility.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds a dedicated page to inform users about the use of the Arcus secret store within Azure Functions. Also adds a link in the general index page so users can be directed directly to it.

Closes #301

### CODE REVIEW

File: docs/preview/features/secret-store/azure-functions.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace MyHttpAzureFunction
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.ConfigureSecretStore(stores =>
+            {
+                stores.AddEnvironmentVariables();
```
Comment (fgheysels) [inline docs/preview/features/secret-store/azure-functions.md:31]: I don't think we need this here, as this is also added on line 34

File: docs/preview/features/secret-store/azure-functions.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+                var keyVaultName = config["KeyVault_Name"];
```
Comment (fgheysels) [inline docs/preview/features/secret-store/azure-functions.md:33]: This code sample gives a nice overview, but should we -for completeness- also show where `config` is coming from by adding a ctor and injecting `IConfiguation` into the `Startup` class ?

**Review by fgheysels — COMMENTED** [review]
Thanks, just a few remarks

File: docs/preview/features/secret-store/azure-functions.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/features/secret-store/azure-functions.md:31]: Ah yes, that's correct. Let's remove this.

File: docs/preview/features/secret-store/azure-functions.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/preview/features/secret-store/azure-functions.md:31]: We should probably use `IFunctionsHostBuilder.GetContext().Configuration` for this. Yes, let's add that.

**Review by fgheysels — APPROVED** [review]

**Pull request merged**